src/components/application_manager/test/commands/mobile/system_request_test.cc

```diff
@@ -114,18 +126,237 @@ class SystemRequestTest
+              IsRequestTypeAllowed(kAppPolicyId, kRequestHttpType))
+      .WillOnce(Return(true));
+
+  am::AppFile file;
+  file.is_download_complete = true;
+  EXPECT_CALL(*mock_app_, GetFile(app_full_file_path)).WillOnce(Return(&file));
+
+  EXPECT_CALL(mock_policy_handler_, ReceiveMessageFromSDK(kFileName, _));
+
+  ExpectManageMobileCommandWithResultCode(mobile_apis::Result::SUCCESS);
+  Run(CreateIVSUMessage());
+}
+
+TEST_F(SystemRequestTest, Run_RequestType_SuccessRequest) {
+  PreConditions();
+
+  EXPECT_CALL(mock_policy_handler_,
+              IsRequestTypeAllowed(kAppPolicyId, kRequestClimate))
+      .WillOnce(Return(true));
+
+  am::AppFile file;
+  file.is_download_complete = true;
+  EXPECT_CALL(*mock_app_, GetFile(app_full_file_path)).WillOnce(Return(&file));
+
+  EXPECT_CALL(app_mngr_,
+              ManageHMICommand(HMIResultCodeIs(
+                  hmi_apis::FunctionID::BasicCommunication_SystemRequest)));
+
+  Run(CreateIVSUMessage(kRequestClimate));
+}
+
+TEST_F(SystemRequestTest, Run_NonExistentApp_ApplicationNotRegisteredResponse) {
+  mock_app_.reset();
+
+  EXPECT_CALL(app_mngr_, application(kConnectionKey))
+      .WillOnce(Return(mock_app_));
+
+  ExpectManageMobileCommandWithResultCode(
+      mobile_apis::Result::APPLICATION_NOT_REGISTERED);
+
+  Run(CreateIVSUMessage());
+}
+
+TEST_F(SystemRequestTest,
+       Run_EmptyParams_DisallowedRequestType_DisallowedResponse) {
+  PreConditions();
+
+  EXPECT_CALL(mock_policy_handler_,
+              IsRequestTypeAllowed(kAppPolicyId, kRequestHttpType))
+      .WillOnce(Return(false));
+
+  ExpectManageMobileCommandWithResultCode(mobile_apis::Result::DISALLOWED);
+
+  Run(CreateIVSUMessage());
+}
+
+TEST_F(SystemRequestTest, Run_InvalidFileNameSyntax_InvalidData) {
+  PreConditions();
+
+  EXPECT_CALL(mock_policy_handler_,
+              IsRequestTypeAllowed(kAppPolicyId, kRequestHttpType))
+      .WillOnce(Return(true));
+
+  ExpectManageMobileCommandWithResultCode(mobile_apis::Result::INVALID_DATA);
+
+  MessageSharedPtr message = CreateIVSUMessage();
+  (*message)[am::strings::msg_params][am::strings::file_name] =
+      kFileName + "\n";
+
+  Run(message);
+}
+
+TEST_F(SystemRequestTest, Run_InvalidFileName_InvalidData) {
+  PreConditions();
+
+  EXPECT_CALL(mock_policy_handler_,
+              IsRequestTypeAllowed(kAppPolicyId, kRequestHttpType))
+      .WillOnce(Return(true));
+
+  ExpectManageMobileCommandWithResultCode(mobile_apis::Result::INVALID_DATA);
+
+  MessageSharedPtr message = CreateIVSUMessage();
+  (*message)[am::strings::msg_params][am::strings::file_name] = kFileName + "/";
+
+  Run(message);
+}
+TEST_F(SystemRequestTest, Run_InvalidSaveBinary_GenericError) {
+  PreConditions();
+
+  EXPECT_CALL(mock_policy_handler_,
+              IsRequestTypeAllowed(kAppPolicyId, kRequestHttpType))
+      .WillOnce(Return(true));
+
+  EXPECT_CALL(app_mngr_,
+              SaveBinary(binary_data_, kSystemFilesPath, kFileName, 0))
```

Comment:
@BSolonenko I don't see any significatn use of binary data. Can we remove it from class members and keep on stack of test cases?